### ISSUE TITLE
Radio button list property data incorrectly converted during migration from 7.15.3 to 8.17.1

### ISSUE THREAD

**Issue by ewuski** [umbraco/Umbraco-CMS#12301]
### Which *exact* Umbraco version are you using? For example: 9.0.1 - don't just write v9

Umbraco 8.17.1

### Bug summary

In Umbraco 7 the property data of the type radio button list is stored as an integers (integer id) of the selected value.

In Umbraco 8 it is stored as a string values of the selected values.

During the migration the integer id is converted into a corresponding string value. The conversion is faulty because it adds a quote around the single value.

### Specifics

In my case a property data value `547` was converted to `"Private"` (with the quote marks) and in this shape the selected value did not show up in the CMS for the page containing the property.
Only after changing it to `Private` (without the quote marks) the value shows selected in the CMS.

### Steps to reproduce

1. Ensure you have a content with Radio button list property selected in a content page.
2. Upgrade from Umbraco 7 version to 8.17.1 (or probably other version 8).
3. Check the content page in the CMS - none of the radio button values is selected.
4. Add a new content page, select the radio value and save.
5. Compare the values of the pages from point 3 and point 4 in the table `umbracoPropertyData` - the first have the property value wrapped in quotation marks, the other does not have them.
6. Update the value with quotation marks in `umbracoPropertyData` table - remove the quotation marks, allow the cache to refresh and check the page in the CMS - the value will be now selected.

The easy fix is to update the values in the database by running a SQL `UPDATE` script eg:

```
UPDATE umbracoPropertyData
SET varcharValue = 'Private'
WHERE varcharValue = '"Private"' AND propertytypeid = 66
```

However the migration conversion is still a bug.

### Expected result / actual result

The Radio button list property values should be migrated to Umbraco 8 without quotes.

**Comment by nul800sebastiaan** [comment]
Thanks for reporting @ewuski - unfortunately we're no longer working on updates for v7 to v8 migrations. Good to see you've found a fix though and by creating a public issue for it other people can now find a solution if they have the same problem 👍

**Comment by ewuski** [comment]
I'm afraid it is also the issue with v9 & v10 (and probably also v8).

The string values in the data type are without the quotation marks, so on assigning the property in the content they are written to the database without the quotation marks either. And in this case the data also shows correctly in the CMS for the content.

However when getting the property value in the API - I receive the values with escaped quotation marks...
So in every case we have extended the partial model and written a custom property that gets the same value but scraps the escaped quotation marks. 

So instead of:

`\"Private\"`

we now correctly receive:

`Private`

So in short there is an issue with storing and reading radio/checkbox string values in v8-10.

**Comment by tormnator** [comment]
We also have the same problem. We're doing a complicated migration from Umbraco 7 to 10, and a migrated radio button list selected value ended up having quotes around the string value. The result is no value is selected in the back office, and any underlying code which tests against the known values fails.

"_... unfortunately we're no longer working on updates for v7 to v8 migrations_". Exactly, it's unfortunate. But why? How many Umbraco 7 (and 8 if the problem is also there) installations are out there still, waiting to be migrated to V10+?

**Comment by marcemarc** [comment]
@tormnator @nul800sebastiaan wondering if this should be added to the documentation 'known issues' with the migration

https://our.umbraco.com/Documentation/Fundamentals/Setup/Upgrading/migrating-to-v8#known-issues

I find it's not so bad if known issues exist if you are aware of them, and can share the workarounds that people need to follow, in order to get to the V10 promised land, rather than everyone getting stuck and frustrated at the same thing.

**Comment by tormnator** [comment]
@marcemarc And/or add it here (found it via your link): https://github.com/umbraco/Umbraco-CMS/issues?utf8=%E2%9C%93&q=label%3Acategory%2Fcontent-migration+

The migration process is complex and time consuming, as in multiple months to do the whole thing for me. It's a big "sell" to be allowed the uninterrupted time to go through it. And for the content migration, it's not a one-time job. The migration has to be documented and/or automated so that it can be done at least twice (both more likely multiple times), once at the start of the project, and once at the very end when all code has been migrated, and you know the complete migration actually can be done. When doing the final migration, it needs to be done fairly quick to avoid unnecessary/lengthy down-time of the old site.

**Comment by marcemarc** [comment]
@tormnator Yes, feel the pain, I've started doing a few days of discovery before attempting one to derisk the project. This involves auditing the property editors and any custom extensions, and running the database migration scripts on various vanilla umbraco version steps, to create a data migration plan, and measure length of time each step takes... once I know I've got a site that will migrate and doesn't have any crazy v7 database index naming issues or orphaned content, only then can I confidently estimate the size of the migration task...
... And sometimes there are better strategies to pure migration, working on a site at the moment where am rebuilding in V10 section by section in parallel to existing site ion a sub domain... Means there isn't that big 'can you test everything at the end big bang project... And we are ending up with a better implemented site as able to utilise newer core stuff, like block list editor.. Instead of ending up with a legacy v7 strategy in v10!

**Comment by tormnator** [comment]
We might be going off topic now, but migration strategy is also a topic which I'd think would be of interest to many, and that I feel would be in Umbraco's best interest to invest some time and resources in providing or facilitating.

I've also created a migration plan. The new site is developed in U10 from scratch using pieces from the U7 code base as needed. Content is migrated from U7 to U8 to U9, and finally to a temporary site in U10 (using built-in and third party migration tools). Then I'm using uSync to bring everything over to the new site piece by piece.

It's very hard, for me at least, to validate the content migration before the content has been brought over to the new site and run with the new code. As I add new code I'm able to bring over content and test, and that's also how I stumbled upon the issue here with the radio button list.

**Comment by sanderwollaert** [comment]
I used the following query:

```UPDATE [umbracoPropertyData]
SET [varcharValue] = TRIM('"' FROM varcharValue) 
WHERE [propertytypeid] IN
(
	SELECT [id]
	FROM [cmsPropertyType]
	WHERE [dataTypeId] IN
	(
		SELECT [nodeId]
		FROM [umbracoDataType]
		WHERE [propertyEditorAlias] = 'Umbraco.RadioButtonList'
		AND [nodeId] > 0
	)
)
AND [varcharValue] IS NOT NULL```

**Comment by emmagarland** [comment]
Thanks for raising this. Came across this same issue during a v7-v13 migration, and can verify that after tweaking @sanderwollaert's query to include the table we want to update (and removing the code tag), the following query resolved it and I can view the populated radio-buttons again:

```
UPDATE [dbo].[umbracoPropertyData]
SET [varcharValue] = TRIM('"' FROM varcharValue) 
WHERE [propertytypeid] IN
(
	SELECT [id]
	FROM [cmsPropertyType]
	WHERE [dataTypeId] IN
	(
		SELECT [nodeId]
		FROM [umbracoDataType]
		WHERE [propertyEditorAlias] = 'Umbraco.RadioButtonList'
		AND [nodeId] > 0
	)
)
AND [varcharValue] IS NOT NULL;
```



If you want to check what will be updated, you can run this first:

```
SELECT * from [dbo].[umbracoPropertyData]
WHERE [propertytypeid] IN
(
	SELECT [id]
	FROM [cmsPropertyType]
	WHERE [dataTypeId] IN
	(
		SELECT [nodeId]
		FROM [umbracoDataType]
		WHERE [propertyEditorAlias] = 'Umbraco.RadioButtonList'
		AND [nodeId] > 0
	)
)
AND [varcharValue] IS NOT NULL;
```

In my case it was thousands of records, so definitely not something I'd want to repick!

#h5yr all!

Emma
